src/common/utils/kaltura-params.js

```diff
@@ -94,28 +102,45 @@ function getReferrer(): string {
 }
 
 /**
- * @param {PKMediaSourceObject} source - source
- * @return {void}
+ * @param {string} url - url
+ * @return {string} - the url with the referrer appended in the query params
  * @private
  */
-function addReferrer(source: PKMediaSourceObject): void {
-  if (source.url.indexOf(REFERRER) === -1) {
-    let delimiter = source.url.indexOf('?') === -1 ? '?' : '&';
+function addReferrer(url: string): string {
+  if (url.indexOf(REFERRER) === -1) {
+    let delimiter = url.indexOf('?') === -1 ? '?' : '&';
     let referrer = btoa(getReferrer().substr(0, 1000));
-    source.url += delimiter + REFERRER + referrer;
+    url += delimiter + REFERRER + referrer;
   }
+  return url;
 }
 
 /**
- * @param {PKMediaSourceObject} source - source
- * @return {void}
+ * @param {string} url - url
+ * @param {PartialKPOptionsObject} playerConfig - player config
+ * @return {string} - the url with the uiconf id appended in the query params
+ * @private
+ */
+function addUIConfId(url: string, playerConfig: PartialKPOptionsObject): string {
+  if (url.indexOf(UICONF_ID) === -1 && playerConfig.session && typeof playerConfig.session.uiConfId === 'number') {
+    let delimiter = url.indexOf('?') === -1 ? '?' : '&';
+    // $FlowFixMe
+    url += delimiter + UICONF_ID + playerConfig.session.uiConfId;
+  }
+  return url;
+}
+
+/**
+ * @param {string} url - url
+ * @return {string} - the url with the client tag appended in the query params
  * @private
  */
-function addClientTag(source: PKMediaSourceObject): void {
-  if (source.url.indexOf(CLIENT_TAG) === -1) {
-    let delimiter = source.url.indexOf('?') === -1 ? '?' : '&';
-    source.url += delimiter + CLIENT_TAG + __VERSION__;
+function addClientTag(url: string): string {
+  if (url.indexOf(CLIENT_TAG) === -1) {
+    let delimiter = url.indexOf('?') === -1 ? '?' : '&';
```

Comment:
I see we have this a lot so better to make this a local method and use in all places

Reply:
done